modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItSqlMultiStatementTxTest.java

```diff
@@ -0,0 +1,219 @@
+import static org.junit.jupiter.api.Assertions.assertThrowsExactly;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.List;
+import org.apache.ignite.internal.sql.BaseSqlMultiStatementTest;
+import org.apache.ignite.internal.tx.InternalTransaction;
+import org.apache.ignite.sql.ExternalTransactionNotSupportedException;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests to verify the execution of queries with transaction control statements.
+ *
+ * @see SqlQueryType#TX_CONTROL
+ */
+@SuppressWarnings("ThrowableNotThrown")
+public class ItSqlMultiStatementTxTest extends BaseSqlMultiStatementTest {
+    /** Number of completed transactions before the test started. */
+    private int numberOfFinishedTransactionsOnStart;
+
+    @BeforeAll
+    void createTable() {
+        sql("CREATE TABLE test (id INT PRIMARY KEY)");
+    }
+
+    @AfterEach
+    void clearTable() {
+        sql("DELETE FROM test");
+    }
+
+    @BeforeEach
+    void saveTxCount() {
+        numberOfFinishedTransactionsOnStart = txManager().finished();
+    }
+
+    void checkFinished(int expected) {
+        assertEquals(numberOfFinishedTransactionsOnStart + expected, txManager().finished());
+    }
+
+    @Test
+    void emptyTransactionControlStatement() {
+        List<AsyncSqlCursor<List<Object>>> cursors = fetchAllCursors(
+                runScript("START TRANSACTION; COMMIT"));
+
+        assertThat(cursors, hasSize(2));
+
+        validateSingleResult(cursors.get(0));
+        validateSingleResult(cursors.get(1));
+
+        checkFinished(1);
+    }
+
+    @Test
+    void basicTxStatements() {
+        fetchAllCursors(runScript(
+                "START TRANSACTION;"
+                        + "INSERT INTO test VALUES(0);"
+                        + "INSERT INTO test VALUES(1);"
+                        + "COMMIT;"
+
+                        + "START TRANSACTION;"
+                        + "COMMIT;"
+
+                        + "START TRANSACTION;"
+                        + "INSERT INTO test VALUES(2);"
+                        + "COMMIT;"
+        ));
+
+        checkFinished(3);
+
+        assertQuery("select count(id) from test")
+                .returns(3L).check();
+    }
+
+    @Test
+    void scriptTransactionRollsBackImplicitly() {
+        {
+            fetchAllCursors(runScript("START TRANSACTION;"));
+            checkNoPendingTransactions();
+            checkFinished(1);
+        }
+
+        {
+            fetchAllCursors(runScript("START TRANSACTION;"
+                    + "INSERT INTO test VALUES(0);"
+                    + "INSERT INTO test VALUES(1);"
+                    + "COMMIT;"
+                    + "START TRANSACTION;"
+                    + "INSERT INTO test VALUES(2);"));
+
+            checkFinished(3);
+
+            checkNoPendingTransactions();
```

Comment:
seems we need documentation issue here, isn`t it ?

Reply:
Yes, definitely :sunglasses:

Reply:
plz link appropriate issue if it already done ?